awx/ui_next/src/screens/Job/JobOutput/HostEventModal.jsx

```diff
@@ -0,0 +1,218 @@
+import React, { useEffect, useState } from 'react';
+import {
+  Button,
+  Modal as PFModal,
+  Tab,
+  Tabs as PFTabs,
+} from '@patternfly/react-core';
+import CodeMirrorInput from '@components/CodeMirrorInput';
+import ContentEmpty from '@components/ContentEmpty';
+import { DetailList, Detail } from '@components/DetailList';
+import { HostStatusIcon } from '@components/Sparkline';
+import { withI18n } from '@lingui/react';
+import { t } from '@lingui/macro';
+import styled from 'styled-components';
+import Entities from 'html-entities';
+
+const entities = new Entities.AllHtmlEntities();
+
+const Modal = styled(PFModal)`
+  --pf-c-modal-box__footer--MarginTop: 0;
+  .pf-c-modal-box__body {
+    overflow-y: hidden;
+  }
+  .pf-c-tab-content {
+    padding: 24px 0;
+  }
+`;
+
+const HostNameDetailValue = styled.div`
+  align-items: center;
+  display: inline-grid;
+  grid-gap: 10px;
+  grid-template-columns: min-content auto;
+`;
+
+const Tabs = styled(PFTabs)`
+  --pf-c-tabs__button--PaddingLeft: 20px;
+  --pf-c-tabs__button--PaddingRight: 20px;
+
+  .pf-c-tabs__list {
+    li:first-of-type .pf-c-tabs__button {
+      &::after {
+        margin-left: 0;
+      }
+    }
+  }
+
+  &:not(.pf-c-tabs__item)::before {
+    position: absolute;
+    top: 0;
+    right: 0;
+    bottom: 0;
+    left: 0;
+    content: '';
+    border-bottom: solid var(--pf-c-tabs__item--BorderColor);
+    border-width: var(--pf-c-tabs__item--BorderWidth) 0
+      var(--pf-c-tabs__item--BorderWidth) 0;
+  }
+`;
+
+function HostEventModal({ handleClose, hostEvent, isOpen, i18n }) {
+  const [hostStatus, setHostStatus] = useState(null);
+  const [activeTabKey, setActiveTabKey] = useState(0);
+
+  useEffect(() => {
+    processEventStatus(hostEvent);
+  }, []);
+
+  const handleTabClick = (event, tabIndex) => {
+    setActiveTabKey(tabIndex);
+  };
+
+  function processEventStatus(event) {
+    let status = null;
+    if (event.event === 'runner_on_unreachable') {
+      status = 'unreachable';
+    }
+    // equiv to 'runner_on_error' && 'runner_on_failed'
+    if (event.failed) {
+      status = 'failed';
+    }
+    // catch the 'changed' case before 'ok', because both can be true
+    if (event.changed) {
+      status = 'changed';
+    }
+    if (
+      event.event === 'runner_on_ok' ||
+      event.event === 'runner_on_async_ok' ||
+      event.event === 'runner_item_on_ok'
+    ) {
+      status = 'ok';
+    }
+    if (event.event === 'runner_on_skipped') {
+      status = 'skipped';
+    }
+    setHostStatus(status);
+  }
```

Comment:
Let's add some consistency (in terms of inputs and outputs) and testability to these three data processing (`"processXYZ"`) functions:

All three functions should accept the value they're transforming as an argument and return the result, e.g:
```js
const StdOut = processStdOutValue(); // before
const stdOut = processStdOutValue(hostEvent.event_data); // after
```

`setHostStatus` and other hook functions are side-effects so lets move them up into the caller, e.g:

```js
  useEffect(() => {
    processEventStatus(hostEvent);
  }, []); // before

  useEffect(() => {
    setHostStatus(processEventStatus(hostEvent));
  }, []); // after
```

Reply:
Cool. I really like this idea. 👍